docs/site/migration/models/remoting-hooks.md

```diff
@@ -6,8 +6,266 @@ sidebar: lb4_sidebar
+  ```
+
+- **Method-level hooks** are executed only for a single method of a given model
+  class.
+
+  These hooks are typically registered on the model class. For example, to run a
+  hook whenever `User.login` is called remotely:
+
+  ```js
+  User.beforeRemote('User.login', handlerFn);
+  ```
+
+LoopBack 4 provides [Interceptors](../../Interceptors.md) feature to enable
+application developers to implement similar functionality. In the following
+sections, we will first explain how to rewrite hook implementations to
+interceptors and then show how to register interceptors to be invoked globally,
+at a class level or at a method level.
+
+## Migrating a hook to an interceptor
+
+Let's take a look at a typical interceptor implementation:
+
+```ts
+async function intercept(
+  invocationCtx: InvocationContext,
+  next: () => ValueOrPromise<InvocationResult>,
+) {
+  try {
+    // Add pre-invocation logic here
+    const result = await next();
+    // Add post-invocation logic here
+    return result;
+  } catch (err) {
+    // Add error handling logic here
+    throw err;
+  }
+}
+```
+
+Every interceptor contains three parts, each corresponding to one of the
+remoting hooks:
+
+- Code from a `beforeRemote` hook belongs to the first section that's executed
+  before the target method is invoked.
+- Code from an `afterRemote` hook belongs to the second section.
+- Code from an `afterRemoteError` hook belongs to the third section inside the
+  `catch` block.
+
+{% include tip.html content=" In LoopBack 3, hooks can be written in one of the
+following three styles:
+
+- A function accepting a callback as the last argument.
+- A function returning a promise, often using
+  [Bluebird](http://bluebirdjs.com/docs/getting-started.html) APIs.
+- An async function using `await` statements for asynchronous flow control.
+
+In LoopBack 4, interceptors are written as async functions using `await`
+statements. "%}
+
+### Example
+
+Consider the following set of remote hooks implementing a very simple request
+logger. We are registering the hooks as global to keep this example simple, but
+the same approach works for model-level and method-level hooks too.
+
+```js
+app.beforeRemote('**', function logBefore(ctx, next) {
+  console.log('About to invoke a method.');
+  next();
+});
+
+app.afterRemote('**', function logAfter(ctx, next) {
+  console.log('Method finished.');
+  next();
+});
+
+app.afterRemoteError('**', function logAfterError(ctx, next) {
+  console.log('Method failed: ', ctx.error);
+});
+```
+
+These three hooks can be converted into a single interceptor.
+
+```ts
+try {
+  // Add pre-invocation logic here
+  console.log('About to invoke a method.');
+  const result = await next();
+  // Add post-invocation logic here
+  console.log('Method finished.');
+  return result;
+} catch (err) {
+  // Add error handling logic here
+  console.log('Method failed: ', err);
+  throw err;
+}
+```
+
+## Migrating global hooks
```

Comment:
Is this part the same as `## Migrating a hook to an interceptor`?  Or is this the second step to show that how to replace the default `intercept` method that generated by `lb4 interceptor` with the converted interceptor we created above?

Reply:
In "Migrating a hook to an interceptor", I am showing how to rewrite the _function body_ of a hook into an interceptor. That part is common for all kinds of hooks (global, model-level, method-level). In this and the following sections, I am showing how to migrate each hook kind (global, model-level, method-level) - how to create the necessary infrastructure (interceptor provider class) and how to register the interceptor at the appropriate place.